common/api/ecschema-metadata.api.md

```diff
@@ -3464,6 +3464,10 @@ export class SchemaKey {
     compareByName(rhs: SchemaKey | string | undefined): boolean;
     compareByVersion(rhs: SchemaKey): number;
     // (undocumented)
+    static fromJson(props: SchemaKeyProps): Promise<SchemaKey>;
+    // (undocumented)
+    static fromJsonSync(props: SchemaKeyProps): SchemaKey;
+    // (undocumented)

```

Comment:
no need for a sync and async version.  You can make it only sync, an async version is needed when the method needs to call another async method, could result in a call out to disk or a call to anything out of process like another service..  So just make one sync method called 'fromJson'.

Also document this method, it may be brief but we will need to have everything documented to come out of alpha/beta so we might as well get started no it now.  Please document everything you're adding in this PR.

Reply:
I added doc comments